app/src/main/java/com/arctouch/codechallenge/data/MovieListDataSource.kt

```diff
@@ -0,0 +1,45 @@
+package com.arctouch.codechallenge.data
+
+import androidx.paging.PageKeyedDataSource
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.model.Movie
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+class MovieListDataSource(private val api: TmdbApi) : PageKeyedDataSource<Long, Movie>() {
+
+    override fun loadInitial(params: LoadInitialParams<Long>, callback: LoadInitialCallback<Long, Movie>) {
+        api.genres(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE)
```

Comment:
What happens if your app face some connections issues?

Reply:
Actually, I'm not sure, probably I'll get an exception from RxJava, because I'm not handling errors. I'm just reusing the code I had, so I'm not sure on the behavior of the error here.